ZuluContent/Skills/Anatomy.cs

```diff
@@ -1,86 +1,96 @@
 using System;
+using Scripts.Zulu.Engines.Classes;
 using Server.Mobiles;
 using Server.Network;
 using Server.Targeting;
+using static Scripts.Zulu.Engines.Classes.SkillCheck;
 
 namespace Server.SkillHandlers
 {
     public class Anatomy
-	{
-		public static void Initialize()
-		{
-			SkillInfo.Table[(int)SkillName.Anatomy].Callback = OnUse;
-		}
+    {
+        public static void Initialize()
+        {
+            SkillInfo.Table[(int) SkillName.Anatomy].Callback = OnUse;
+        }
 
-		public static TimeSpan OnUse( Mobile m )
-		{
-			m.Target = new InternalTarget();
+        public static TimeSpan OnUse(Mobile m)
+        {
+            m.Target = new InternalTarget();
 
-			m.SendLocalizedMessage( 500321 ); // Whom shall I examine?
+            m.SendLocalizedMessage(500321); // Whom shall I examine?
 
-			return TimeSpan.FromSeconds( 1.0 );
-		}
+            return Configs[SkillName.Anatomy].Delay;
+        }
 
-		private class InternalTarget : Target
-		{
-			public InternalTarget() :  base ( 8, false, TargetFlags.None )
-			{
-			}
+        private class InternalTarget : Target
+        {
+            public InternalTarget() : base(-1, false, TargetFlags.None)
+            {
+            }
 
-			protected override void OnTarget( Mobile from, object targeted )
-			{
-				if ( from == targeted )
-				{
-					from.LocalOverheadMessage( MessageType.Regular, 0x3B2, 500324 ); // You know yourself quite well enough already.
-				}
-				else if ( targeted is TownCrier )
-				{
-					((TownCrier)targeted).PrivateOverheadMessage( MessageType.Regular, 0x3B2, 500322, from.NetState ); // This person looks fine to me, though he may have some news...
-				}
-				else if ( targeted is BaseVendor && ((BaseVendor)targeted).IsInvulnerable )
-				{
-					((BaseVendor)targeted).PrivateOverheadMessage( MessageType.Regular, 0x3B2, 500326, from.NetState ); // That can not be inspected.
-				}
-				else if ( targeted is Mobile )
-				{
-					Mobile targ = (Mobile)targeted;
+            protected override void OnTarget(Mobile from, object targeted)
+            {
+                if (!(targeted is Mobile))
+                {
+                    from.SendAsciiMessage(33, "This has no anatomy at all!");
+                    return;
+                }
 
-					int marginOfError = Math.Max( 0, 25 - (int)(from.Skills[SkillName.Anatomy].Value / 4) );
+                if (!(from as IShilCheckSkill)?.CheckSkill(SkillName.Anatomy, -1,
```

Comment:
Nitpick: If you're not aware you can explicitly check for true/false (the result is a `bool?`)
```csharp
(from as IShilCheckSkill)?.CheckSkill(SkillName.Anatomy, -1, SkillName.Anatomy].DefaultPoints) == false
```

Reply:
Mmm yes it's returning a bool, I'm not sure why doing an explicit check is better tho? More readable?

Reply:
Yeah just for readability, they work the same.